### ISSUE TITLE
Research queue is not well synchronized

### ISSUE THREAD

**Issue by graille** [hubastard/nebula#363]
When the host is making a research queue, the client will see it until the first research is completed. After that, the queue will disapear from client screen but the research will continue on the host side. Client will be notified when a research finish on host side, but will not see any ongoing research on his taskbar (for the client, all the game act like no technology is being researched).

This can conduct to a bug when the client can launch a research on his side which will not appear on host side. In this case, host and client will be working on two different technologies. The client have to reconnect to resync his technology tree with the host.

In my game, we "fixed" that by agreed that only the host can manage the research queue. But the client cannot see the current technology queue and have to ask to the host to know what's going on.

**Comment by EL-S** [comment]
I also constantly experience this bug, having the affected clients disconnect and reconnect works as a temporary fix

**Comment by Needix** [comment]
Anyone already working here? I would try to investigate this, if not

**Comment by PhantomGamers** [comment]
> Anyone already working here? I would try to investigate this, if not

I think @hubastard found a discrepancy between the host and client sync code because he said he fixed something relating to this in #394

**Comment by hubastard** [comment]
Yes the issue that I fixed in this PR was that for one of the packet, we were not broadcasting the packet to other clients when the packet author was another client.

**Comment by ST-DDT** [comment]
Today we encountered this bug/a similar bug with v0.3.1.
The client was way faster with completing the research queue.
In most cases, the client even had progress, when nobody had the required materials.
The same rarely happened on the server where cancelling and requeuing a tech added ~25% of the missing materials.
We had to fix these client desyncs multiple times by reconnecting the client.

Aside from that there were rarely any issues in our MP session today.
Thanks for creating/mainting this awesome mod.
